### ISSUE TITLE
@DolphinTestRunner Capability 

### ISSUE THREAD

**Issue by montanajava** [canoo/dolphin-platform#1]
It would be very nice to have a @DolphinTestRunner that would allow us to test components, something like this:

```
@DolphinTestRunner
public class TestDolphinStuff() {

      // here  model is injected
      @DolphinModel
      CalculatorModel calculatorModel;

    @Test
   public void testSomething() {
       // here I have access to the implementation of the model.
       calculatorModel.setFirstValue(1);
       // or can instantiate my own
       calculatorModel = DolphinTestRunner.defaultInstance(CalculatorModel.class); // or something like that.

       // now I can test real stuff. Especially handy if the model is not entirely dumb, i.e. just a DTO.
   }
}
```

The test runner would start up the necessary container services and support injection.

This is a first stab at what would be nice, and only touches on the the features of the DolphinFramework, now that I am playing around with it.

It should support JUnit given its market share.

**Comment by hendrikebbers** [comment]
Hi,
we will add test support soon ;) Currently our main goal is to release a first version with Spring / JavaEE support on the server and JavaFX / JS clients. Once this is done we hope to get some feedback to finalize APIs. I think it don't make sense to create a full test support before that.

**Comment by hendrikebbers** [comment]
A Spring support is planed for 0.9

**Comment by hendrikebbers** [comment]
see #80 and #81

**Comment by hendrikebbers** [comment]
Closed since we have a test API for controller tests (at least for Spring)
